packages/workflow/src/Interfaces.ts

```diff
@@ -1012,6 +1039,23 @@ export interface INodePropertyCollection {
 	values: INodeProperties[];
 }
 
+export interface INodePropertyValueExtractorBase {
+	type: string;
+}
+
+export interface INodePropertyValueExtractorRegex extends INodePropertyValueExtractorBase {
+	type: 'regex';
+	regex: string | RegExp;
```

Comment:
1. Curious if we can we type this as `RegExp` instead of `string | RegExp` so TS disallows strings that are invalid regex strings?
2. If we can, do we need `type: 'regex'` if we can already derive it from the type of `regex`?

Reply:
I should actually remove the RegExp option. At the moment all the validation is done client side so that'll actually stop it from working.